ga4gh/repo_manager.py

```diff
@@ -0,0 +1,383 @@
+        destPath = os.path.join(
+            self._repoPath, self.readsDirName, fileName)
+        self._assertPathEmpty(destPath, inRepo=True)
+        os.mkdir(destPath)
+        fileDestPath = os.path.join(destPath, os.path.basename(filePath))
+        os.rename(filePath, fileDestPath)  # TODO add option for copy / link 
+        # TODO index reference set
+        self._repoEmit("ReferenceSet '{}' added".format(fileName))
+
+    def addReadGroupSet(self, datasetName, filePath):
+        """
+        Add a read group set to the repo
+        """
+        self._check()
+        self._checkDataset(datasetName)
+        self._checkFile(filePath, self.bamExtension)
+        fileName = filenameWithoutExtension(filePath)
+        destPath = os.path.join(
+            self._repoPath, self.datasetsDirName, datasetName,
+            self.readsDirName, fileName)
+        self._assertPathEmpty(destPath, inRepo=True)
+        os.rename(filePath, destPath)  # TODO add option for copy / link
+        # TODO index bam
+        self._repoEmit("ReadGroupSet '{}' added to dataset '{}'".format(
+            fileName, datasetName))
+
+    def addVariantSet(self, datasetName, filePath):
+        """
+        Add a variant set to the repo
+        """
+        self._check()
+        self._checkDataset(datasetName)
+        self._checkFolder(filePath, self.vcfExtension)
+        dirName = os.path.basename(filePath)
+        destPath = os.path.join(
+            self._repoPath, self.datasetsDirName, datasetName,
+            self.variantsDirName, dirName)
+        self._assertPathEmpty(destPath, inRepo=True)
+        os.rename(filePath, destPath)  # TODO add option for link
+        # TODO index vcf files
+        self._repoEmit("VariantSet '{}' added to dataset '{}'".format(
+            dirName, datasetName))
+
+    def list(self):
+        """
+        List the contents of the repo
+        """
+        self._check()
+        self._repoEmit("Listing")
+
+        repoObj = self.listObjs()
+        self._emit(self.referenceSetsDirName)
+        for referenceSet in repoObj._referenceSets:
+            self._emitIndent(referenceSet)
+            for reference in referenceSet._references:
+                self._emitIndent(reference, 2)
+        self._emit(self.datasetsDirName)
+        for dataset in repoObj._datasets:
+            self._emitIndent(dataset)
+            self._emitIndent(self.readsDirName, 2)
+            for readGroupSet in dataset._readGroupSets:
+                self._emitIndent(readGroupSet, 3)
+            self._emitIndent(self.variantsDirName, 2)
+            for variantSet in dataset._variantSets:
+                self._emitIndent(variantSet, 3)
+                for variant in variantSet._variants:
+                    self._emitIndent(variant, 4)
+        
+    def listObjs(self):
+        """
+        Return the contents of the repo as an object
+        """
+        repoObj = Repository(self._repoPath)
+
+        # referenceSets
```

Comment:
OK, we're definitely recreating code here. Right now, we should be able to do something like
```
backend = FileSystemBackend(self._repoPath)
for dataset in backend.getDatasets():
   # etc.
```
We need to figure out what the distinction between a 'Backend' and a 'Repository' is. I guess a Backend _has_ a Repository? This would need a refactor of the Backend class to take out all of the ``addReferenceSet`` type methods and point them at the ``Repository`` object instead.

Does this make sense? I.e., we would have code like
```
repo = FileSystemRepository(self._repoPath)
for dataset in repo.getDatasets():
   # etc.
```